### PR TITLE
feat(xcp-ng-generic): BFD and Valgrind are now optional

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -61,16 +64,27 @@ set(CMAKE_THREAD_PREFER_PTHREAD YES)
...
+  if (Bfd_FOUND)
+    list(APPEND LIBS Bfd::Bfd)
+    list(APPEND FIND_MODULES cmake/FindBfd.cmake)
+  endif ()
+endif ()
+
+if (ENABLE_VALGRIND)
+  find_package(Valgrind)
+  if (Valgrind_FOUND)
+    set(HAVE_VALGRIND 1)
```
Comment (benjamreis) [inline CMakeLists.txt:84]: Is it necessary? Since you have `Valgrind_FOUND` can't you use it instead?

File: config/XcpNgGenericConfig.cmake.in [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
-  set(CMAKE_THREAD_PREFER_PTHREAD YES)
-  find_dependency(Threads REQUIRED)
+    if (@Valgrind_FOUND@)
+      include("${XCP_CMAKE_DIR}/FindValgrind.cmake")
+    endif ()
 
+    include(CMakeFindDependencyMacro)
+    set(CMAKE_THREAD_PREFER_PTHREAD YES)
+    find_dependency(Threads REQUIRED)
+  endif ()
```
Comment (benjamreis) [inline config/XcpNgGenericConfig.cmake.in:16]: Comment to what if it corresponds

File: config/XcpNgGenericConfig.cmake.in [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
   include("${XCP_CMAKE_DIR}/@XCP_TARGETS_FILE@")
 endif ()
```
Comment (benjamreis) [inline config/XcpNgGenericConfig.cmake.in:18]: same

**Review by benjamreis — CHANGES_REQUESTED** [review]

**Comment by benjamreis** [comment]
Generally: comment the `endif() # toto` with corresponding `if(toto)` for clarity.
I find it especially usefull with `cmake` code. :)

File: CMakeLists.txt (hunk 1 above)
Comment (Wescoeur) [reply CMakeLists.txt:84]: I must declare a new variable for a C file:
https://github.com/xcp-ng/xcp-ng-generic-lib/blob/master/src/coroutine.c#L34

File: config/XcpNgGenericConfig.cmake.in (hunk 2 above)
Comment (Wescoeur) [reply config/XcpNgGenericConfig.cmake.in:16]: ?

File: config/XcpNgGenericConfig.cmake.in (hunk 3 above)
Comment (Wescoeur) [reply config/XcpNgGenericConfig.cmake.in:18]: Same "?".

**Comment by Wescoeur** [comment]
What's the purpose? You don't make the same thing for C or python code. :)

**Comment by benjamreis** [comment]
C and Python doesn't have `endif()`. I treat more like the `#ifdef` and `#endif` which I comment similarly.
But it's just a question of preference it's not mandatory of course. ^^

**Comment by Wescoeur** [comment]
Yes but a condition like this:
```C
if (toto) {
    ...
}
```
is similar to:
```cmake
if (toto)
    ...
endif ()
```
In the two cases, the same control flow is used. I suppose I comment C preprocessor out of habit or for historical reasons. :thinking:

**Pull request merged**